packages/user/AuthDyn/service/src/lib.rs

```diff
@@ -0,0 +1,258 @@
+    fn isRejectSys(
+        sender: AccountNumber,
+        authorizers: Vec<AccountNumber>,
+        auth_set: Option<Vec<AccountNumber>>,
+    ) -> bool {
+        is_auth(sender, authorizers, auth_set, false)
+    }
+
+    fn is_auth_other(
+        auth_service: AccountNumber,
+        sender: AccountNumber,
+        authorizers: Vec<AccountNumber>,
+        auth_set: Vec<AccountNumber>,
+        is_approval: bool,
+    ) -> bool {
+        use psibase::services::auth_delegate::action_structs;
+
+        let service_caller = ServiceCaller {
+            sender: Wrapper::SERVICE,
+            service: auth_service,
+            flags: 0,
+        };
+
+        if is_approval {
+            service_caller.call(
+                action_structs::isAuthSys::ACTION_NAME.into(),
+                action_structs::isAuthSys {
+                    sender,
+                    authorizers,
+                    auth_set: Some(auth_set),
+                },
+            )
+        } else {
+            service_caller.call(
+                action_structs::isRejectSys::ACTION_NAME.into(),
+                action_structs::isRejectSys {
+                    sender,
+                    authorizers,
+                    auth_set: Some(auth_set),
+                },
+            )
+        }
+    }
+
+    fn is_auth(
+        sender: AccountNumber,
+        authorizers: Vec<AccountNumber>,
+        auth_set: Option<Vec<AccountNumber>>,
+        is_approval: bool,
+    ) -> bool {
+        // Make sure we're not in a loop
+        let mut auth_set = auth_set.unwrap_or_default();
+        if auth_set.contains(&sender) {
+            return false;
+        }
+        auth_set.push(sender);
+
+        let dynamic_auth_policy = Policy::get_assert(sender).dynamic_policy();
+
+        match dynamic_auth_policy {
+            DynamicAuthPolicy::Single(single_auth) => {
+                authorizers.contains(&single_auth.authorizer)
+                    || is_auth_other(
+                        Accounts::call().getAuthOf(single_auth.authorizer),
+                        single_auth.authorizer,
+                        authorizers,
+                        auth_set,
+                        is_approval,
+                    )
+            }
+            DynamicAuthPolicy::Multi(mut multi_auth) => {
+                check(
+                    multi_auth.threshold != 0,
+                    "multi auth threshold cannot be 0",
+                );
+
+                let total_possible_weight = multi_auth
+                    .authorizers
+                    .iter()
+                    .fold(0, |acc, authorizer| acc + authorizer.weight);
+
+                let required_weight = if is_approval {
+                    multi_auth.threshold
+                } else {
+                    // Anti-threshold: total - threshold + 1
+                    total_possible_weight
+                        .saturating_sub(multi_auth.threshold)
+                        .saturating_add(1)
```

Comment:
This is wrong when `threshold > total_possible_weight`

Reply:
Is this okay with 

```
check(
    total_possible_weight >= required_weight,
    "multi auth returned insufficient weight",
);
```  
afterwards?